### DISCUSSION TITLE
Return value from usePrepareContractWrite does not make sense

### DISCUSSION THREAD

**Discussion by holma91** [wevm/wagmi#865] [General]
I seriously struggle to see why x.write is a write function but `y.write === undefined` in the snippet down below. Everything is the same except from the address in the args array.
```js
const { config: setApprovalForAllConfigX } = usePrepareContractWrite({
  addressOrName: mumbai.predictionMarket,
  contractInterface: PredictionMarket.abi,
  functionName: 'setApprovalForAll',
  args: ['0x5078E54E37C98BDF2624f671cbeC46dF62467EAB', true],
});

const x = useContractWrite(setApprovalForAllConfigX);

console.log('x', x.write);

const { config: setApprovalForAllConfigY } = usePrepareContractWrite({
  addressOrName: mumbai.predictionMarket,
  contractInterface: PredictionMarket.abi,
  functionName: 'setApprovalForAll',
  args: ['0xdcb9048D6bb9C31e60af7595ef597ADC642B9cB6', true],
});

const y = useContractWrite(setApprovalForAllConfigY);

console.log('y', y.write);
```

What am I missing here? If I use the same address in both X and Y, then y.write is also a valid function.

**Comment by jxom** [comment]
Are there any errors coming back in the prepare phase? You can check this with the `status` and `error` return values on `usePrepareContractWrite`.
Could be possible that the address isn't valid for that function call?

**Reply by holma91** [reply]
> Are there any errors coming back in the prepare phase? You can check this with the `status` and `error` return values on `usePrepareContractWrite`. Could be possible that the address isn't valid for that function call?

No I don't think so since the same function (setApprovalForAll) is called on the same contract (PredictionMarket which is an ERC721) in both cases. The only difference is the argument for the "operator" parameter in setApprovalForAll. I tested with `status` and `error` now and `status === idle` and `error === null`.

This is just an example though, my main problem is that I want to call a function where the arguments to it are not known when I call usePrepareContractWrite (they get populated by the user in the UI). So my idea is to have the user input in state, and have these state variables in the args array. That doesn't work though because the `write` function I get back is undefined.

**Reply by holma91** [reply]
does the format of the "mock object" that I use in the inital args array be in the exact correct format?
